src/libYARP_OS/include/yarp/os/Vocab.h

```diff
@@ -53,23 +36,45 @@ namespace yarp {
  * traverse the network will be human readable/writable.
  *
  */
-class YARP_OS_API yarp::os::Vocab {
-public:
+namespace yarp {
+namespace os {
+// We need a constexpr for efficient switching.
+// Use as, for example, makeVocab('s','e','t')
+constexpr yarp::conf::vocab32_t createVocab(char a, char b = 0, char c = 0, char d = 0)  {
+    return ((((yarp::conf::vocab32_t)(d))<<24)+(((yarp::conf::vocab32_t)(c))<<16)+(((yarp::conf::vocab32_t)(b))<<8)+((yarp::conf::vocab32_t)(a)));
```

Comment:
A bit cleaner:

```c++
return ((yarp::conf::vocab32_t)d << 24) + ((yarp::conf::vocab32_t)c << 16) + ((yarp::conf::vocab32_t)b << 8) + (yarp::conf::vocab32_t)a;
```

I'd remove those parentheses around each parameter, the final `return`, and each C-style cast ([ref](https://en.cppreference.com/w/cpp/language/operator_precedence)), here and at `VOCABX` below, as an unnecessary remnant of the former implementation via C preprocessor macros.

Reply:
What would you think about c++-style casts, which are more explicit?

Reply:
Also I'd really appreciate if you could expand this on more than one line, and add them in the proper order (i.e. a-b-c-d, not d-c-b-a) in order to make it more readable, like here:

https://github.com/robotology-playground/sharedlibpp/blob/45b5cae5b50c6431e5eae8bb8aebf6d751054f8b/src/shlibpp/SharedLibraryClassApi.h#L63:L67